src/main/java/com/conveyal/datatools/manager/utils/MergeFeedUtils.java

```diff
@@ -0,0 +1,144 @@
+
+import static com.conveyal.gtfs.loader.Field.getFieldIndex;
+
+public class MergeFeedUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(MergeFeedUtils.class);
+
+    /**
+     * Get the ids (e.g., trip_id, service_id) for the provided table from the zipfile.
+     */
+    public static Set<String> getIdsForTable(ZipFile zipFile, Table table) throws IOException {
+        Set<String> ids = new HashSet<>();
+        String keyField = table.getKeyFieldName();
+        CsvReader csvReader = table.getCsvReader(zipFile, null);
+        if (csvReader == null) {
+            LOG.warn("Table {} not found in zip file: {}", table.name, zipFile.getName());
+            return ids;
+        }
+        Field[] fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
+        // Get the key field (id value) for each row.
+        int keyFieldIndex = getFieldIndex(fieldsFoundInZip, keyField);
+        while (csvReader.readRecord()) ids.add(csvReader.get(keyFieldIndex));
+        csvReader.close();
+        return ids;
+    }
+
+    public static String stopCodeFailureMessage(int stopsMissingStopCodeCount, int stopsCount, int specialStopsCount) {
+        return String.format(
+            "If stop_code is provided for some stops (for those with location_type = " +
+                "empty or 0), all stops must have stop_code values. The merge process " +
+                "found %d of %d total stops that were incorrectly missing stop_code values. " +
+                "Note: \"special\" stops with location_type > 0 need not specify this value " +
+                "(%d special stops found in feed).",
+            stopsMissingStopCodeCount,
+            stopsCount,
+            specialStopsCount
+        );
+    }
+
+    /**
+     * Collect zipFiles for each feed version before merging tables.
+     * Note: feed versions are sorted by first calendar date so that future dataset is iterated over first. This is
+     * required for the MTC merge strategy which prefers entities from the future dataset over past entities.
+     */
+    public static List<FeedToMerge> collectAndSortFeeds(Set<FeedVersion> feedVersions) {
+        return feedVersions.stream().map(version -> {
+            try {
+                return new FeedToMerge(version);
+            } catch (Exception e) {
+                LOG.error("Could not create zip file for version: {}", version.version);
+                return null;
+            }
+        }).filter(Objects::nonNull).filter(entry -> entry.version.validationResult != null
+            && entry.version.validationResult.firstCalendarDate != null)
+            // MTC-specific sort mentioned in above comment.
+            // TODO: If another merge strategy requires a different sort order, a merge type check should be added.
+            .sorted(Comparator.comparing(entry -> entry.version.validationResult.firstCalendarDate,
+                Comparator.reverseOrder())).collect(Collectors.toList());
```

Comment:
The indentation makes this hard to read. Here's my recommended formatting:

```suggestion
        return feedVersions.stream()
            .map(version -> {
                try {
                    return new FeedToMerge(version);
                } catch (Exception e) {
                    LOG.error("Could not create zip file for version: {}", version.version);
                    return null;
                }
            })
            .filter(Objects::nonNull)
            .filter(
                entry -> entry.version.validationResult != null && 
                    entry.version.validationResult.firstCalendarDate != null
            )
            // MTC-specific sort mentioned in above comment.
            // TODO: If another merge strategy requires a different sort order, a merge type check should be added.
            .sorted(
                Comparator.comparing(
                    entry -> entry.version.validationResult.firstCalendarDate, 
                    Comparator.reverseOrder()
                )
            )
            .collect(Collectors.toList());
```